create.ps1

```diff
@@ -1,35 +1,190 @@
-$success = $false;
-$filePath = "\\server\c$\HelloID\StudentPasswords.csv";
-
-$mutexName = 'helloIDWriteback'
-$mutex = New-Object 'Threading.Mutex' $false, $mutexName
-		
-try{
-	[void]$mutex.WaitOne(6000);
-	#add new row
-	$newRow = New-Object -TypeName PsObject -Property (@{ID=$p.ExternalId
-														Email=$p.Accounts.ActiveDirectory.Mail
-                                                      })
-    if(-Not($dryRun -eq $True)){  
-	    $newRow | Export-Csv -NoTypeInformation -Path $filePath -Append
-    }
-    $success = $true;
-    $aRef = $newRow.Email;
-    $account = $newRow;
+#################################################
+# HelloID-Conn-Prov-Target-CSV-Create
+# Create and update or correlate to csv row
+# PowerShell V2
+#################################################
+# Enable TLS1.2
+[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12
+
+# Set debug logging
+switch ($actionContext.Configuration.isDebug) {
+    $true { $VerbosePreference = "Continue" }
+    $false { $VerbosePreference = "SilentlyContinue" }
+}
+$InformationPreference = "Continue"
+$WarningPreference = "Continue"
+
+#region functions
+#endregion functions
+
+#region account
+# Define correlation
+$correlationField = $actionContext.CorrelationConfiguration.accountField
+$correlationValue = $actionContext.CorrelationConfiguration.accountFieldValue
+
+$account = [PSCustomObject]$actionContext.Data
+#endRegion account
+
+try {
+    #region Verify correlation configuration and properties
+    $actionMessage = "verifying correlation configuration and properties"
+
+    if ($actionContext.CorrelationConfiguration.Enabled -eq $true) {
+        if ([string]::IsNullOrEmpty($correlationField)) {
+            throw "Correlation is enabled but not configured correctly."
+        }
+    
+        if ([string]::IsNullOrEmpty($correlationValue)) {
+            throw "The correlation value for [$correlationField] is empty. This is likely a mapping issue."
+        }
+    }
+    else {
+        throw "Correlation is disabled. However, this connector requires correlation, as it's designed to support only one object per person."
+    }
+    #endregion Verify correlation configuration and properties
+
+    #region Import CSV data
+    $actionMessage = "importing data from CSV file at path [$($actionContext.Configuration.CsvPath)]"
+    
+    $csvContent = $null
+    $csvContent = Import-Csv -Path $actionContext.Configuration.CsvPath -Delimiter $actionContext.Configuration.Delimiter -Encoding $actionContext.Configuration.Encoding
+    
+    # Group on correlation field to match employee to CSV row(s)
+    $csvContentGrouped = $csvContent | Group-Object -Property $correlationField -AsString -AsHashTable
+
+    Write-Verbose "Imported data from CSV file at path [$($actionContext.Configuration.CsvPath)]. Result count: $(($csvContent | Measure-Object).Count)"
```

Comment:
I would suggest using `Write-Information` for verbose logging. That will also work when running on-premises. However, do note that, informational messages will only be shown when running in preview mode.